### PR TITLE
[Landcover Toolkit] Add multiband support to medioid composites

### PR DESCRIPTION
**Pull request by gorelick-google**

### CODE REVIEW

File: toolkits/landcover/impl/composites.js [hunk 1]
```diff
@@ -81,41 +81,45 @@ function createTemporalComposites(
...
-  
-  // Add a band containing the difference between the index band value for each
-  // pixel and the median of the index band across all pixels in the collection.
-  var mosaic = collection.map(function(img) {
-    var diff = median.subtract(img.select(indexBand)).abs();
-    var index = ee.Image(0).subtract(diff);
-    return img.addBands(index.rename('medioid_index_'));
-  }).qualityMosaic('medioid_index_');
+function createMedioidComposite(collection, bands) {
+  bands = bands === undefined ? ".*" : bands;
```
Comment (gino-m) [inline toolkits/landcover/impl/composites.js]: Nit: Let's use `'` string delimiter for consistency.

Comment (gino-m) [reply toolkits/landcover/impl/composites.js]: Also, `=== undefined` and ternary aren't necessary - you can simply do;
`bands = bands || '.*'`

File: toolkits/landcover/impl/composites.js [hunk 2]
```diff
@@ -81,41 +81,45 @@ function createTemporalComposites(
...
- * and returning the complete observation where that band is closest
- * to the median value.
+ * Computes a medioid composite by finding the median and returning the
+ * complete observation closest to the median value as measured by
+ * euclidiean distance over the specified band(s).
  *
  * @param {!ee.ImageCollection} collection The collection to composite.
- * @param {string|number} indexBand The name of the band which to
- *    select the median value. If omitted, the first band is used.
+ * @param {Array<string|number>|string|number} bands A band identifier
```
Comment (gino-m) [inline toolkits/landcover/impl/composites.js]: If this arg is optional should be:
`@param {(Array<string|number>|string|number)=}`

File: toolkits/landcover/impl/dataset.js [hunk 3]
```diff
@@ -110,17 +110,18 @@ Dataset.prototype.createTemporalComposites = function(
...
- * Computes a medioid composite by finding the median of one band
- * and returning the complete observation where that band is closest
- * to the median value.
+ * Computes a medioid composite by finding the median and returning the
+ * complete observation closest to the median value as measured by
+ * euclidiean distance over the specified band(s).
  *
- * @param {string} band The name of the band which to
- *    select the median value.
+ * @param {Array<string|number>|string|number} bands A band identifier
```
Comment (gino-m) [inline toolkits/landcover/impl/dataset.js]: Also here:
`@param {(Array<string|number>|string|number)=}`

**Comment by gino-m** [comment]
Re checks, it appears GitHub is still trying to pull from #43:
image

Something probably got weird because there are two PRs against one branch. I'll override the checks on merge this time.

File: toolkits/landcover/impl/composites.js (hunk 1 above)
Comment (gorelick-google) [reply toolkits/landcover/impl/composites.js]: Done

File: toolkits/landcover/impl/composites.js (hunk 2 above)
Comment (gorelick-google) [reply toolkits/landcover/impl/composites.js]: Done

File: toolkits/landcover/impl/dataset.js (hunk 3 above)
Comment (gorelick-google) [reply toolkits/landcover/impl/dataset.js]: Done

File: toolkits/landcover/impl/composites.js (hunk 1 above)
Comment (gorelick-google) [reply toolkits/landcover/impl/composites.js]: Not true.
Bands can be 0 and the || would result in selecting all bands instead of band 0.

**Pull request merged**